### PR TITLE
Refactor(eos_designs): Remove wan_transit and rename wan_edge to wan_router

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

**THIS IS BREAKING** but the features are still marked as preview so we can implement this in a minor release.

- Rename the node type `wan_edge` to `wan_router`
- Remove the node type `wan_transit` and `cv_pathfinder_role` from the schema
  Instead use a new key at the node level:
  ```
  node.cv_pathfinder_transit_mode = [zone, region] <default=none>
  ```

## Component(s) name

`arista.avd.eos_designs`

## How to test

molecule updated

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 1]
```diff
@@ -369,9 +366,27 @@ def wan_ha_flow_tracker_name(self: SharedUtils) -> str:
         """
         return "WAN-FLOW-TRACKER"
 
+    # TODO should we keep this or change it?
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]: no should just be `is_cv_pathfinder_client`

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 2]
```diff
@@ -63,17 +63,14 @@ def wan_listen_ranges(self: SharedUtils) -> list:
         )
 
     @cached_property
-    def cv_pathfinder_role(self: SharedUtils) -> str | None:
+    def cv_pathfinder_transit_mode(self: SharedUtils) -> str | None:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]: I think we could simplify some other checks by returning None here also for the `"none"` case.

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 3]
```diff
@@ -369,9 +366,27 @@ def wan_ha_flow_tracker_name(self: SharedUtils) -> str:
...
+        """
+        Return the CV Pathfinder role based on the wan_role (server or client) and the
+        node.cv_pathfinder_transit_mode key.
+        """
+        if self.wan_mode != "cv-pathfinder" or not self.is_wan_router:
+            return None
+
+        if self.is_wan_server:
+            return "pathfinder"
+        elif self.cv_pathfinder_transit_mode == "none":
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]: Something about this match makes me feel uncomfortable. Suggested None value for the `"none"` case.

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/node-type-wan-configuration.md [hunk 4]
```diff
@@ -55,14 +55,11 @@
...
+        # only when the `wan_mode` root key is set to `cv_pathfinder`.
 
-        # This key is used for Pathfinder designs only when the `wan_mode` root
-        # key is set to `cv_pathfinder`.
-
-        # `pathfinder` is only a valid if `wan_role` is `server`.
-        # `edge` and `transit region` are only valid if `wan_role` is `client`.
-        cv_pathfinder_role: <str; "edge" | "transit region" | "pathfinder">
+        # 'zone' is not supported today.
+        cv_pathfinder_transit_mode: <str; "none" | "zone" | "region"; default="none">
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/docs/tables/node-type-wan-configuration.md]: Would it all be simpler if we dropped the "none" value here and just did not have a default value?

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml [hunk 5]
```diff
@@ -443,3 +439,6 @@ application_classification:
         port_values:
           - 30000-30100
           - 31000-31100
+
+# wan_transit:
+#   blah: test_da_stuff
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml]:
```suggestion
```

**Review by gmuloc — COMMENTED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested in POC

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**